src/renderers/react/externalStore.ts

```diff
@@ -0,0 +1,70 @@
+import type { ReactElement } from "react";
+import { useEffect, useState } from "react";
+
+/**
+ * Creates a store to allow state within React to be programmatically updated
+ * outside of a React context.
+ */
+export const createStore = <Value>(getValue: Value) => {
```

Comment:
I think this is a really eloquent solution. As discussed when pairing, there may be a more fitting name.